dotnet/samples/TeamsChefBot/Program.cs

```diff
@@ -0,0 +1,81 @@
+using TeamsChefBot;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.TeamsAI;
+using Microsoft.TeamsAI.AI.Planner;
+using Microsoft.TeamsAI.State;
+using Microsoft.TeamsAI.AI.Prompt;
+using Microsoft.TeamsAI.AI;
+using Microsoft.TeamsAI.AI.Moderator;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddHttpClient("WebClient", client => client.Timeout = TimeSpan.FromSeconds(600));
+builder.Services.AddHttpContextAccessor();
+
+// Prepare Configuration for ConfigurationBotFrameworkAuthentication
+var config = builder.Configuration.Get<ConfigOptions>();
+builder.Configuration["MicrosoftAppType"] = "MultiTenant";
+builder.Configuration["MicrosoftAppId"] = config.BOT_ID;
+builder.Configuration["MicrosoftAppPassword"] = config.BOT_PASSWORD;
+
+// Create the Bot Framework Authentication to be used with the Bot Adapter.
+builder.Services.AddSingleton<BotFrameworkAuthentication, ConfigurationBotFrameworkAuthentication>();
+
+// Create the Cloud Adapter with error handling enabled.
+// Note: some classes expect a BotAdapter and some expect a BotFrameworkHttpAdapter, so
+// register the same adapter instance for all types.
+builder.Services.AddSingleton<CloudAdapter, AdapterWithErrorHandler>();
+builder.Services.AddSingleton<IBotFrameworkHttpAdapter>(sp => sp.GetService<CloudAdapter>());
+builder.Services.AddSingleton<BotAdapter>(sp => sp.GetService<CloudAdapter>());
+
+builder.Services.AddSingleton<IStorage, MemoryStorage>();
+builder.Services.AddSingleton<AzureOpenAIPlannerOptions>(_ => new AzureOpenAIPlannerOptions(config.AZURE_OPENAI_API_KEY, "gpt-4-32k", config.AZURE_OPENAI_ENDPOINT));
```

Comment:
Open question - use OpenAI or AzureOpenAI or both in sample?

Reply:
Good question, the same question on the moderator.

Reply:
Is there a reason you are using the `gpt-4-32` model? Last I checked it's not an available model on Azure Open AI. It would be best to stick with `text-davinci-003` or `gpt-3.5-turbo`.

Reply:
I think we should use OpenAI's planner and moderator for most samples as it's more accessible to the general public. Right now, a regular user would need special access an Azure OpenAI resource

Reply:
Updated to use `text-davinci-003`.
Commented out Azure OpenAI.